app-tests/gromacs/gromacs_check.py

```diff
@@ -0,0 +1,117 @@
+import reframe as rfm
+from hpctestlib.sciapps.gromacs.benchmarks import gromacs_check
+
+
+@rfm.simple_test
+class lumi_gromacs_check(gromacs_check):
+    maintainers = ['mszpindler']
+    use_multithreading = False
+    executable_opts += ['-dlb yes', '-ntomp 1', '-npme -1']
+    valid_prog_environs = ['cpeGNU']
+
+    num_nodes = parameter([1, 2, 4], loggable=True)
+    allref = {
+        1: {
+            'gfx90a': { 
+            },
+            'zen3': { 
+                'HECBioSim/Crambin': (320.0, None, None, 'ns/day'),
+                'HECBioSim/Glutamine-Binding-Protein': (115.503, None, None, 'ns/day'),  
+                'HECBioSim/hEGFRDimer': (18.938, None, None, 'ns/day'),
+                'HECBioSim/hEGFRDimerSmallerPL': (33.785, None, None, 'ns/day'),
+                'HECBioSim/hEGFRDimerPair': (7.759, None, None, 'ns/day'),
+            },
+        },
+        2: {
+            'gfx90a': { 
+            },
+            'zen3': { # Collected initial performance numbers after SS11 upgrade
+                'HECBioSim/Crambin': (280.267, None, None, 'ns/day'),
+                'HECBioSim/Glutamine-Binding-Protein': (210.0, None, None, 'ns/day'),
+                'HECBioSim/hEGFRDimer': (31.0, None, None, 'ns/day'),
+                'HECBioSim/hEGFRDimerSmallerPL': (30.571, None, None, 'ns/day'),
+                'HECBioSim/hEGFRDimerPair': (14.269, None, None, 'ns/day'),
+            },
+        },
+        4: {
+            'gfx90a': { 
+            },
+            'zen3': { # Collected initial performance numbers after SS11 upgrade
+                'HECBioSim/Crambin': (295.263, None, None, 'ns/day'),
+                'HECBioSim/Glutamine-Binding-Protein': (190.459, None, None, 'ns/day'),
+                'HECBioSim/hEGFRDimer': (32.232, None, None, 'ns/day'),
+                'HECBioSim/hEGFRDimerSmallerPL': (34.619, None, None, 'ns/day'),
+                'HECBioSim/hEGFRDimerPair': (21.66, None, None, 'ns/day'),
+                'HECBioSim/hEGFRtetramerPair': (9.58, None, None, 'ns/day'),
+            },
+        },
+    }
+
+    @run_after('init')
+    def overwrite_cmds(self):
+    # Need to overwrite because of downloads being not possible on compute nodes
+        self.prerun_cmds = []
+        self.executable_opts = ['-nb', self.nb_impl, '-s', 
+            f'{self.bench_name}/benchmark.tpr'
+        ]
+
+    @run_after('init')
+    def setup_modules(self):
+        if self.nb_impl == 'cpu':
+            self.modules = ['GROMACS']
+            self.time_limit = '15m'
```

Comment:
This could be moved to the class body since probably the upcoming gpu version will use the a gromacs module with the same name. It shouldn't be necessary to decide based on the type of run.

Reply:
I wouldn't say so, see at easyconfigs generating modules with suffix, like this one: `GROMACS/2021.5-cpeGNU-21.12-CPU`.

Reply:
What I mean is that probably the gpu test will use the default name of the module as well, since the config will take care of loading the corresponding `partition` module. I think that setting `modules = ['GROMACS']` would work for both variants.

Reply:
You are right.